plugins/database/mysql/connection_producer.go

```diff
@@ -13,12 +13,21 @@ import (
 	"sync"
 	"time"
 
+	cloudmysql "cloud.google.com/go/cloudsqlconn/mysql/mysql"
+
 	"github.com/go-sql-driver/mysql"
 	"github.com/hashicorp/go-secure-stdlib/parseutil"
 	"github.com/hashicorp/go-uuid"
+	"github.com/mitchellh/mapstructure"
+
 	"github.com/hashicorp/vault/sdk/database/helper/connutil"
 	"github.com/hashicorp/vault/sdk/database/helper/dbutil"
-	"github.com/mitchellh/mapstructure"
+)
+
+const (
+	authTypeIAM   = connutil.AuthTypeIAM
```

Comment:
Nit: I might wonder what IAM is referring to in this context if looking at this with fresh eyes. Perhaps we can add GCP to these context to make it more obvious?

Reply:
Yeah, I think that's a good call